index.js

```diff
@@ -91,6 +99,29 @@ class Store {
     return this;
   }
 
+  /**
+   * Assign `value` to `key` while retaining prior members of `value` if
+   * `value` is a map.
+   *
+   * ```js
+   * store.set('a', { b: c });
+   * //=> {a: { b: c }}
+   *
+   * store.append('a', { d: e });
+   * //=> {a: { b: c, d: e }}
+   * ```
+   *
+   * @name .append
+   * @param {String} `key`
+   * @param {any} `val` The value to append to `key`. Must be a valid JSON type: String, Number, Array or Object.
+   * @return {Object} `Store` for chaining
+   * @api public
+   */
+
+  append(key, val) {
+    return this.set(key, Object.assign(this.get(key) || {}, val));
```

Comment:
Should additional checks be done to ensure the value from `this.get(key)` is an object?

Reply:
Done. It now explicitly ensures that `val` is an object (and not null and not an array). If not an object, `.merge` acts like `.set`.